### PR TITLE
Enable bundle format on 3.1.2 branch and switch CRD to v1

### PR DESCRIPTION
**Pull request by salv-orlando**
This PR cherry-picks PR#136 and PR#128 to the 3.1.2 branch.
References to operator version and image tag are fixed accordingly.

In addition, the ncpInstall CR is  moved from to the v1 group from the v1beta1 group

### CODE REVIEW

File: bundle/generate_bundle.sh [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+# build kubernetes bundle image
+docker build -t ${BUNDLE_REPO}/${BUNDLE_IMG_NAME}-k8s:${BUNDLE_VERSION} -f ./kubernetes/bundle.Dockerfile ./kubernetes
+docker push ${BUNDLE_REPO}/${BUNDLE_IMG_NAME}-k8s:${BUNDLE_VERSION}
+opm index add --build-tool docker --bundles ${BUNDLE_REPO}/${BUNDLE_IMG_NAME}-k8s:${BUNDLE_VERSION} --tag ${BUNDLE_REPO}/ncp-operator-index-k8s:${BUNDLE_VERSION}
+
+# build openshift4 bundle image
+docker build -t ${BUNDLE_REPO}/${BUNDLE_IMG_NAME}-oc:${BUNDLE_VERSION} -f ./openshift4/bundle.Dockerfile ./openshift4
+docker push ${BUNDLE_REPO}/${BUNDLE_IMG_NAME}-oc:${BUNDLE_VERSION}
+opm index add --build-tool docker --bundles ${BUNDLE_REPO}/${BUNDLE_IMG_NAME}-oc:${BUNDLE_VERSION} --tag ${BUNDLE_REPO}/ncp-operator-index-oc:${BUNDLE_VERSION}
+popd
```
Comment (jwsui) [inline bundle/generate_bundle.sh:76]: "no newline" warning here, maybe i forgot this in previous patch.

File: bundle/kubernetes/manifests/nsx-container-plugin-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+                  valueFrom:
+                    fieldRef:
+                      fieldPath: metadata.name
+                - name: OPERATOR_NAME
+                  value: nsx-ncp-operator
+                - name: NCP_IMAGE
+                  value: nsx-ncp:latest
+                - name: WATCH_NAMESPACE
+                  value: nsx-system-operator
+                image: docker.io/vmware/nsx-container-plugin-operator:v3.1.2.1
```
Comment (jwsui) [inline bundle/kubernetes/manifests/nsx-container-plugin-operator.clusterserviceversion.yaml:106]: Just want to confirm the image tag is `v3.1.2.1`?

Comment (salv-orlando) [reply bundle/kubernetes/manifests/nsx-container-plugin-operator.clusterserviceversion.yaml:106]: Yes, we published a new release to ensure wait_for_security_policy_sync was disabled.

I might need to push another tag as image might need to be rebuilt to get latest version of package (to pass security checks).

In the meanwhile we can go with this tag.

File: bundle/openshift4/manifests/nsx-container-plugin-operator.clusterserviceversion.yaml [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+                  valueFrom:
+                    fieldRef:
+                      fieldPath: metadata.name
+                - name: OPERATOR_NAME
+                  value: nsx-ncp-operator
+                - name: NCP_IMAGE
+                  value: nsx-ncp:latest
+                - name: WATCH_NAMESPACE
+                  value: nsx-system-operator
+                image: docker.io/vmware/nsx-container-plugin-operator:v3.1.2.1
```
[the remark this answers is not in the document]
Comment (salv-orlando) [reply bundle/openshift4/manifests/nsx-container-plugin-operator.clusterserviceversion.yaml:116]: np!

**Comment by jwsui** [comment]
LGTM, thanks.

**Pull request closed without merging**